R/model.R

```diff
@@ -405,7 +405,9 @@ compile_method <- function(quiet = TRUE,
   } else {
     exe_base <- file.path(dir, basename(self$stan_file()))
   }
-  exe <- cmdstan_ext(paste0(strip_ext(exe_base), exe_suffix))
+  current_hash = digest::digest(self$code(),algo='xxhash64')
+  exe_name_without_hash = paste0(strip_ext(exe_base),exe_suffix)
+  exe <- cmdstan_ext(paste0(exe_name_without_hash,'_',current_hash))
   model_name <- sub(" ", "_", paste0(strip_ext(basename(self$stan_file())), "_model"))
```

Comment:
This should replace self$stan_file() with exe_name_without_hash otherwise the model name will be bernoulli_hash_model, which is not what we want.

Reply:
Sorry, missed this comment until now. I coincidentally just posted [a related issue](https://github.com/stan-dev/cmdstanr/issues/337). If we determine there that model_name should always use the exe, I'll change things here to comply.